daphne/src/testing.rs

```diff
@@ -195,32 +209,103 @@ impl DapAggregator<BearerToken> for MockAggregator {
 impl DapHelper<BearerToken> for MockAggregator {
     async fn mark_aggregated(
         &self,
-        _task_id: &Id,
-        _report_shares: &[ReportShare],
+        task_id: &Id,
+        report_shares: &[ReportShare],
     ) -> Result<HashMap<Nonce, TransitionFailure>, DapError> {
-        // Return empty HashMap (for now).
-        // TODO(nakatsuka-y) Implement correct functionality.
-        let early_fails: HashMap<Nonce, TransitionFailure> = HashMap::new();
-        return Ok(early_fails);
+        let mut early_fails: HashMap<Nonce, TransitionFailure> = HashMap::new();
+
+        for report_share in report_shares.iter() {
+            let bucket_info = BucketInfo::new(
+                self.tasks
+                    .get(task_id)
+                    .ok_or_else(|| DapError::fatal("no task found"))?,
+                task_id,
+                &report_share.nonce,
+            );
+
```

Comment:
It looks like you acquire the lock on report_store, then release it up to three times. It would be easier to read (and probably more efficient) to acquire once:
```rust
              let mut report_store = self.report_store.lock().expect("hella");
              if !report_store.contains_key(&bucket_info) {
                  report_store.insert(bucket_info.clone(), ReportStore::new());
              }

              match report_store
                  .get_mut(&bucket_info)
                  .ok_or_else(|| DapError::fatal("no report store found"))?
                  .process_put_processed(report_share.nonce.clone())
              {
                  Ok(()) => (),
                  Err(failure_reason) => {
                      early_fails.insert(report_share.nonce.clone(), failure_reason);
                  }
              }
```

Reply:
Thank you for the suggestion, I completely forgot about this.
I think this makes the code look much cleaner.